### DISCUSSION TITLE
node:internal/fs/utils:348 throw err; ^ Error: EPERM: operation not permitted

### DISCUSSION THREAD

**Discussion by vincemammoliti** [spatie/browsershot#665] [Q&A] [answered]
I'm trying to get this package working with a simple example, but I can't find a solution to the following error.
I am using Windows 11, IIS, Laravel 9, Node.js v18.7.0. Puppeteer has been installed to the project.
```
Symfony \ Component \ Process \ Exception \ ProcessFailedException
The command "C:^\nodejs^\node.exe ^"C:^\Users^\me^\Documents^\sandbox^\my-app^\vendor^\spatie^\browsershot^\src/../bin/browser.js^" ^"^{^\^"url^\^":^\^"https:^\/^\/example.com^\^",^\^"action^\^":^\^"pdf^\^",^\^"options^\^":^{^\^"path^\^":^\^"C:^\^\Users^\^\me^\^\Documents^\^\sandbox^\^\my-app^\^\storage^\^\app^\/public^\/test.pdf^\^",^\^"args^\^":^[^],^\^"viewport^\^":^{^\^"width^\^":800,^\^"height^\^":600^}^}^}^"" failed. Exit Code: 1(General error) Working directory: C:\Users\me\Documents\sandbox\my-app\public Output: ================ Error Output: ================ node:internal/fs/utils:348 throw err; ^ Error: EPERM: operation not permitted, lstat 'C:\Users\me' at Object.realpathSync (node:fs:2552:7) at toRealPath (node:internal/modules/cjs/loader:405:13) at Module._findPath (node:internal/modules/cjs/loader:561:22) at resolveMainPath (node:internal/modules/run_main:19:25) at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:75:24) at node:internal/main/run_main_module:17:47 { errno: -4048, syscall: 'lstat', code: 'EPERM', path: 'C:\\Users\\me' } Node.js v18.7.0
```

This is the simple example I am trying to get working:
```
use Spatie\Browsershot\Browsershot;

class DownloadController extends Controller
{
    public function __invoke()
    {
        Browsershot::url('https://example.com')
            ->setNodeBinary('C:\nodejs\node.exe')
            ->save(storage_path('app/public/test.pdf'));

        return "Success";
    }
}
```

Can anyone please help me with this?

**Comment by vincemammoliti** [comment, marked as answer]
This is resolved now. The issue was related to Node and IIS config/directory permissions.

Despite setting up the correct permissions to run my Laravel application from C:\Users{username}{subfolders}\my-app\ , when running node.js as the user identity associated with the IIS application pool it caused a permissions error on the C:\Users{username}\ directory.

After moving my application code from my Documents directory to C:\inetpub\apps\my-app, and setting up the same IIS user permissions, it is now working.

For reference, this answer on a different problem helped me get to that solution: https://stackoverflow.com/a/68937635

**Reply by carolina-CortezFreudenthal** [reply]
![Screenshot 2022-11-18 at 12 54 00](https://user-images.githubusercontent.com/113165054/202706783-91176c9a-9f23-4792-9325-110d2fbe071d.png)

hi!, I have the same error on ,but i can't fix
